invenio_app_ils/circulation/indexer.py

```diff
@@ -47,10 +47,23 @@ def index_referenced_records(loan):
     # be wrong and corrected at the second re-indexing.
     loan_class = current_circulation.loan_record_cls
     loan_record = loan_class.get_record_by_pid(loan["pid"])
+
     referenced.append(
         dict(pid_type=CIRCULATION_LOAN_PID_TYPE, record=loan_record)
     )
 
+    # add all the other loans, as after indexing this one, they
+    # will be affected in search
+    pending_loans = \
+        document.search_loan_references().filter("term", state="PENDING") \
```

Comment:
This should be enough:
```suggestion
        document.search_loan_references() \
```